rnote-ui/src/workspacebrowser/filerow/actions/rename.rs

```diff
@@ -0,0 +1,101 @@
+use std::path::PathBuf;
+
+use gtk4::{
+    gio, glib,
+    glib::clone,
+    pango,
+    prelude::FileExt,
+    traits::{ButtonExt, EditableExt, PopoverExt, StyleContextExt, WidgetExt, BoxExt},
+    Align, Button, Entry, Label, Popover
+};
+
+use gettextrs::gettext;
+
+use crate::workspacebrowser::{filerow::widget_helper, FileRow};
+
+impl FileRow {
+    pub fn rename_action(&self) -> gio::SimpleAction {
+        let rename_action = gio::SimpleAction::new("rename-file", None);
+
+        rename_action.connect_activate(clone!(@weak self as filerow => move |_action_rename_file, _| {
+            if let Some(current_file) = filerow.current_file() {
+                if let Some(current_path) = current_file.path() {
+                    if let Some(parent_path) = current_path.parent().map(|parent_path| parent_path.to_path_buf()) {
+                        let entry = get_entry(&current_path);
+                        let label = get_label();
+                        let (apply_button, popover) = widget_helper::entry_dialog::get_entry_dialog(&entry, &label);
+
+                        filerow.menubutton_box().append(&popover);
+
+                        connect_entry(&entry, &apply_button, parent_path.clone());
+                        connect_apply_button(&apply_button, &popover, &entry, parent_path.clone(),
+                            current_file.clone());
+
+                        popover.popup();
+                    }
+                }
+            }
+        }));
+
+        rename_action
+    }
+}
+
+fn get_entry(current_path: &PathBuf) -> Entry {
```

Comment:
These methods should be called `create_ ...` to indicate a constructor, not `get_ ...` which is commonly associated for getting existing properties in objects

Reply:
done